tests/testenv.sh

```diff
@@ -16,30 +15,84 @@ for i in "$@"
 			exit 1
 			;;
 		x86)
-			export CoreRT_BuildArch=x86
+			CoreRT_BuildArch=x86
 			;;
 		x64)
-			export CoreRT_BuildArch=x64
+			CoreRT_BuildArch=x64
 			;;
 		arm)
-			export CoreRT_BuildArch=arm
+			CoreRT_BuildArch=arm
 			;;
 		arm64)
-			export CoreRT_BuildArch=arm64
+			CoreRT_BuildArch=arm64
 			;;
 		dbg)
-			export CoreRT_BuildType=Debug
+			CoreRT_BuildType=Debug
 			;;
 		debug)
-			export CoreRT_BuildType=Debug
+			CoreRT_BuildType=Debug
 			;;
 		rel)
-			export CoreRT_BuildType=Release
+			CoreRT_BuildType=Release
 			;;
 		release)
-			export CoreRT_BuildType=Release
+			CoreRT_BuildType=Release
 			;;
 		*)
 			;;
 	esac
 done
+
+if [ -z ${CoreRT_BuildArch} ]; then
+    echo "Set CoreRT_BuildArch to x86/x64/arm/arm64"
+    exit -1
+fi
+
+if [ -z ${CoreRT_BuildType} ]; then
+    echo "Set CoreRT_BuildType to Debug or Release"
+    exit -1
+fi
+
+# Use uname to determine what the OS is.
+OSName=$(uname -s)
+case $OSName in
+    Darwin)
+        CoreRT_BuildOS=OSX
+        ;;
+
+    FreeBSD)
+        CoreRT_BuildOS=FreeBSD
+        ;;
+
+    Linux)
+        CoreRT_BuildOS=Linux
+        ;;
+
+    NetBSD)
+        CoreRT_BuildOS=NetBSD
+        ;;
+
+    *)
+        echo "Unsupported OS $OSName detected, configuring as if for Linux"
+        CoreRT_BuildOS=Linux
+        ;;
+esac
+
+export CoreRT_BuildArch
+export CoreRT_BuildType
+export CoreRT_BuildOS
+
+__build_os_lowcase=$(echo "${CoreRT_BuildOS}" | tr '[:upper:]' '[:lower:]')
+if [ ${__build_os_lowcase} != "osx" ]; then
+    __BuildRid=ubuntu.14.04-${CoreRT_BuildArch}
+else
+    __BuildRid=osx.10.10-${CoreRT_BuildArch}
+fi
+
+export CoreRT_AppDepSdkPkg=toolchain.${__BuildRid}.Microsoft.DotNet.AppDep
+export CoreRT_AppDepSdkVer=1.0.6-prerelease-00004
+
+__ScriptDir=$(cd "$(dirname "$0")"; pwd -P)
+
```

Comment:
Not your change, but `$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)` should get the right sh.